### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by Avemark**
Most importantly adding of a separator in `Healthcare#to_s` and making use of `#to_s` in `Record#to_ai`

### CODE REVIEW

File: lib/gs1/record.rb [hunk 1]
```diff
@@ -48,8 +47,8 @@ def to_ai
     end
 
     def ==(other)
-      self.class == other.class &&
-        to_s == other.to_s
+      self.class.equal?(other.class) &&
```
Comment (stefanahman) [inline lib/gs1/record.rb]: Please explain this change

File: spec/gs1/record_spec.rb [hunk 2]
```diff
@@ -67,19 +77,15 @@ class RecordSuperDummy < Record; end
       let(:other_record) { GS1::RecordDummy.new(other_data) }
 
       context 'with same data' do
-        let(:other_data) { data }
+        let(:other_data) { data.dup }
 
-        it 'is equal' do
-          is_expected.to eq(true)
-        end
+        it { is_expected.to eq(other_record) }
```
Comment (stefanahman) [inline spec/gs1/record_spec.rb]: This generates a non-descriptive output:
```
#==
  when same class
    with same data
      should eq #<GS1::RecordDummy:0x007f8064418a98 @data="data">
    with different data
      should not eq #<GS1::RecordDummy:0x007f8062973a08 @data="not data">
  when different class
    with same data
      should not eq #<GS1::RecordSuperDummy:0x007f8062970fd8 @data="data">
    with different data
      should not eq #<GS1::RecordSuperDummy:0x007f806297bf78 @data="not data">
```

File: spec/support/define_length_matcher.rb [hunk 3]
```diff
@@ -21,6 +21,8 @@
 
   def actual_allowed
     actual.definitions[:length][:allowed]
+  rescue NoMethodError
```
Comment (stefanahman) [inline spec/support/define_length_matcher.rb]: This is not tested? And why is it needed?

File: lib/gs1/record.rb (hunk 1 above)
Comment (Avemark) [reply lib/gs1/record.rb]: from commit message: 

equal?() tests object identity as well.
== sometimes includes type conversion where eql?() does not.

it was part of the feedback from mutation testing

**Pull request merged**